### PR TITLE
Make mocks movable

### PR DESCRIPTION
**Pull request by marbub**
If you use templated mocks it can be very handy to create them in
factory functions. To archive it the mocks needs to be movable, which is
prevented by the definition of ~FunctionMockerBase which removes the move
contructor. So we added a default move constructor again.

mock.h
```
class MockSqliteWriteStatement
{
public:
    MOCK_METHOD2(write,
                 void (Utils::SmallStringView));

};
```
implementation.h
```
template<typename WriteStatement>
class SqliteStatementFactory
{
private:
    static WriteStatement createStatement()
    {
        SqlStatementBuilder builder;
        builder.setTableName("newSybols");

        return WriteStatement(builder.sqlStatement());
    }

    WriteStatement m_statement{createStatement()};
}
```

### CODE REVIEW

**Comment by marbub** [comment]
I signed it!

**Comment by marbub** [comment]
I tried to set my email address as alternative email address but it still does not match the CLA.

File: googlemock/include/gmock/gmock-spec-builders.h [hunk 1]
```diff
@@ -1464,6 +1464,9 @@ class FunctionMockerBase : public UntypedFunctionMockerBase {
   typedef typename Function<F>::ArgumentMatcherTuple ArgumentMatcherTuple;
 
   FunctionMockerBase() : current_spec_(this) {}
+#if GTEST_LANG_CXX11
+  FunctionMockerBase(FunctionMockerBase&&) = default;
```
Comment (nyibbang) [inline googlemock/include/gmock/gmock-spec-builders.h]: Visual Studio 2013 does not support defaulted move constructor/operator.

Comment (marbub) [reply googlemock/include/gmock/gmock-spec-builders.h]: There is no feature testing macro. So we should exclude Visual Studio 2013?

Comment (nyibbang) [reply googlemock/include/gmock/gmock-spec-builders.h]: It would be easier to implement it instead of using "= default" I think.

**Comment by rburny** [comment]
Unfortunately, this will not work with the current GMock implementation.
GMock maintains the map of objects, stored by their address:

https://github.com/google/googletest/blob/a325ad2db5deb623eab740527e559b81c0f39d65/googlemock/src/gmock-spec-builders.cc#L549

This is used by `AllowUninterestingCalls`, `VerifyAndClearExpectations`, and other internal methods. If you move the mock object, these would stop working (in particular, Nice and StrictMock would not work correctly).

File: googlemock/include/gmock/gmock-spec-builders.h (hunk 1 above)
Comment (KindDragon) [reply googlemock/include/gmock/gmock-spec-builders.h]: `GTEST_LANG_CXX11` is not defined for Visual Studio 2013, so it's ok

**Comment by gennadiycivil** [comment]
Closing , this approach wont work

**Pull request closed without merging**